packages/env/lib/wordpress.js

```diff
@@ -91,7 +91,13 @@ async function configureWordPress( environment, config ) {
 	);
 
 	// Set wp-config.php values.
-	for ( const [ key, value ] of Object.entries( config.config ) ) {
+	for ( let [ key, value ] of Object.entries( config.config ) ) {
+		// Ensure correct port setting from config when configure WP urls.
+		if ( key === 'WP_SITEURL' || key === 'WP_HOME' ) {
```

Comment:
I wonder if we should introduce a separate function for "merging" or "overriding" values in the config.

Reply:
I like this idea. Especially for deep merging of .wp-env.json and .wp-env.override.json objects this would be nice.

Reply:
I looked at adding a separate function in https://github.com/WordPress/gutenberg/pull/22568 (for some different config values) so we can probably skip it here :)